packages/destination-actions/src/destinations/intercom/identifyContact/index.ts

```
@@ -0,0 +1,187 @@
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+
+interface IntercomSearchData {
+  total_count: number
+  data: Array<IntercomContact>
+}
+
+interface IntercomContact {
+  id: string
+}
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Identify Contact',
+  description: 'Create or Update a Contact.',
+  defaultSubscription: 'type = "identify"',
+  platform: 'web',
+  fields: {
+    role: {
+      type: 'string',
+      required: true,
+      description: 'The role of the contact. Accepted values are `user` or `lead`. Can only be updated if `lead`.',
+      label: 'Role',
+      default: 'lead'
+    },
+    external_id: {
+      type: 'string',
+      description: 'A unique identifier generated outside Intercom. Required if role=user and email is blank.',
+      label: 'External ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    email: {
+      type: 'string',
+      description: "The contact's email. Required if role=user and external_id is blank.",
+      label: 'Email',
+      format: 'email',
+      default: {
+        '@path': '$.traits.email'
+      }
+    },
+    phone: {
+      label: 'Phone Number',
+      description: "The contact's phone number.",
+      type: 'string',
+      default: {
+        '@path': '$.traits.phone'
+      }
+    },
+    name: {
+      type: 'string',
+      description: "The contact's name.",
+      label: 'Name',
+      default: {
+        '@path': '$.traits.name'
+      }
+    },
+    avatar: {
+      label: 'Avatar',
+      description: 'URL of image to be associated with contact profile.',
+      type: 'string',
+      format: 'uri',
+      default: {
+        '@path': '$.traits.avatar'
+      }
+    },
+    signed_up_at: {
+      label: 'Signed Up At',
+      type: 'datetime',
+      description: 'The timestamp when the contact was created.',
+      default: {
+        '@path': '$.createdAt'
```

I don't know if I would use a default for this field at all. `createdAt` is the timestamp the current event was created on the client device.

---

^ Agreed. I think we should remove any default for contact creation time as we don't know that this was their first identify call. The contact could already exist and we would then be overriding their true creation date.

---

Also seems like this should be in UNIX timestamp. I see we clean up timestamps for trackEvent, but I don't see us checking that the identifyContact timestamps are unix strings?

---

@kdaswani seems that track threw errors when the unix timestamp wasn't passed, but it was silently ignored on identify! Will fix this now, thanks for the catch!